### PR TITLE
Challenge Solution PR

### PR DESCRIPTION
**Pull request by jalmarazg**
Added basic functionality and structure to retrieve Gifs and manage loading state.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+                        to = "/favorite"
+                        exact>
+                        Favorites</NavLink></li>
+                    </ul>
+                </nav>
+            </header>
+            <Switch>
+              <Route path = "/home" exact component = {GifPage} />
+              <Route path = "/favorite" exact component = {GifFavorite} />
+              <Redirect from = "/" to = "/home" />
```
Comment (bgilm) [inline src/App.js]: It is a good idea to have a redirect from every other non existing route point to a NotFound or to the index with a message.

Comment (bgilm) [inline src/App.js]: 👍

File: src/api/api.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+const apiURL = "http://api.giphy.com";
+const apiTrends = "/v1/gifs/trending";
+const apiSearch = "/v1/gifs/search";
+const apiKey = "MKGAXNQl5cXUBSBMrXSsufVZ9bqvhX6p"
```
Comment (bgilm) [inline src/api/api.js]: Use `.env` variables to store your api keys, also it is a good practice to use CAPS for static vars. Dont version that file and provide an example for its usage.

https://www.npmjs.com/package/dotenv

File: src/store/localStorage.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Special thanks to: Dan Abramov
+// For the awesome "LocalStorage Persistence" Tutorial that helped me implement it on the challenge
```
Comment (bgilm) [inline src/store/localStorage.js:2]: 👍

File: src/App.styled.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import styled from 'styled-components';
+import { NavLink } from 'react-router-dom';
+
+export const GifSection = styled.div`
+  text-align: center;
+  background-color: ${props => props.theme.gifPageBackground};
```
Comment (timgivois) [inline src/App.styled.js:6]: 👏

File: src/UI/test/SearchBar.test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import SearchBar from '../SearchBar'
```
Comment (timgivois) [inline src/UI/test/SearchBar.test.js:2]: can you test the component instead of the conttainer?

**Comment by jalmarazg** [comment]
Travis keeps sending errors (Which I had until the following):
- My workaround to test locally was downloading "jest" and "jest-cli" 
- After installing them I ran "yarn jest" 
- Got, at the moment of my last commit, the following: 

percentages

File: src/reducers/searchValues.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import searchActions from '../actions/searchValues'
```
Comment (Robfz) [inline src/reducers/searchValues.js:1]: I think this reducer could be split in two. Even if all the actions are defined in `searchActions`, some of them are concerned with I/O and others with the search value.

File: src/components/GifFavorite/GifFavorite.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    updateState = (event) => {
+        if(event === ""){
+            this.setState({searchResults: this.props.favoritedImages})
+        } else {
+            this.setState({ searchedValue: event })
+        }
+    }
+
+    handleSearch = () => {
+        if(this.state.searchedValue === ""){
```
Comment (Robfz) [inline src/components/GifFavorite/GifFavorite.js:35]: This filtering should be done at the Store level.

File: src/App.js [hunk 8]
```diff
@@ -1,19 +1,59 @@
...
+        <StyledDiv>   
+              <StyledHeader {...this.props}>
+                <StyledUl>
+                    <StyledLi><StyledNavLink 
+                    to = "/"
+                    activeClassName = "act"
+                    exact>
+                    Home </StyledNavLink></StyledLi>
+
+                    <StyledLi><StyledNavLink
```
Comment (bgilm) [inline src/App.js:39]: 👍

File: src/components/GifCards/GifCards.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+                </GifCardsStyled>);
+            });
+
+            if(arrays.length === 0){
+                arrays = <StyledH1> No Results Found :(. Go search and fav some awesome gifs! </StyledH1>;
+            }
+        }
+
+        return(
+            <Fragment>
```
Comment (bgilm) [inline src/components/GifCards/GifCards.js:32]: 👍

File: src/ErrorBoundary/ErrorBoundary.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+
+class ErrorBoundary extends Component {
```
Comment (bgilm) [inline src/ErrorBoundary/ErrorBoundary.js:3]: 👍

File: src/App.styled.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+export const StyledDiv = styled.div`
+  text-align: center;
+  padding: 0px;
+  border: 0px;
+  margin: -10px;
+  width: 100%;
+  height: 100%;
+  position: absolute;
+`
+/*
```
Comment (timgivois) [inline src/App.styled.js:31]: delete unused code mate

**Comment by timgivois** [comment]
to solve your problem, delete your `package-lock.json` and ran `yarn jest`.

**Pull request closed without merging**